File: Week_02/242/valid_anagram.go
package main

func isAnagram(s string, t string) bool {
	counter := make([]int, 26)

	for _, c := range s {
		counter[c-'a']++
	}
	for _, c := range t {
		counter[c-'a']--
	}

	for _, i := range counter {
		if i != 0 {
			return false
		}
	}
	return true
}

File: Week_02/242/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        counter = [0] * 26

        for c in s:
            counter[ord(c) - ord('a')] += 1
        for c in t:
            counter[ord(c) - ord('a')] -= 1

        for i in counter:
            if i != 0:
                return False
        return True
